etna/ciim/client.py

```diff
@@ -1,636 +1,70 @@
-    """
-
-    EVIDENTIAL = "evidential"
-    INTERPRETIVE = "interpretive"
-
-
-class SortBy(StrEnum):
-    """Options for sorting /search results by a given field."""
-
-    RELEVANCE = ""
-    TITLE = "title"
-    DATE_CREATED = "dateCreated"
-    DATE_OPENING = "dateOpening"
-
+from etna.core.api import JSONAPIClient
 
-class SortOrder(StrEnum):
-    """Options for sort order for /search results."""
+DEFAULT_REFERENCE_NUMBER = "[unknown]"
+DEFAULT_SUMMARY_TITLE = "[unknown]"
+DEFAULT_IAID = None
 
-    ASC = "asc"
-    DESC = "desc"
 
-
-class Template(StrEnum):
-    """@template block to include with response.
-
-    Supported by all endpoints.
+class CIIMClient(JSONAPIClient):
     """
-
-    DETAILS = "details"
-    RESULTS = "results"
-
-
-def prepare_filter_aggregations(
-    items: Sequence[str] | None,
-) -> list[str] | None:
-    """
-    Filter format in items: 'field:value', 'field:value:or'
-    Prepares i.e. removes/replaces special chars from a filter fields' value to be passed to the api
-    When using filter with multiple values, specific fields require OR operator to be specified,
-    otherwise AND is used by default.
-
-    Example:
-    before-prepare: "heldBy:Birmingham: Archives, Heritage and Photography Service"
-    after-prepare:  "heldBy:Birmingham Archives Heritage and Photography Service"
-    before-prepare: "heldBy:Staffordshire and Stoke-on-Trent Archive Service: Staffordshire County Record Office"
-    after-prepare:  "heldBy:Staffordshire and Stoke on Trent Archive Service Staffordshire County Record Office"
-    Special char single quote i.e. ' is not prepared
-    before-prepare: "heldBy:Labour History Archive and Study Centre (People's History Museum/University of Central Lancashire)"
-    after-prepare:  "heldBy:Labour History Archive and Study Centre People's History Museum University of Central Lancashire "
+    Client for interacting with the CIIM API.
     """
-    if not items:
-        return None
 
-    regex = r"([/():,\&\-\|+@!.])"
-    subst = " "
-    field_list_to_prepare = ["heldBy"]
-    filter_prepared_list = []
-    fields_using_or_operator = ["heldBy", "collection", "level"]
+    def __init__(self, api_url: str = f"{settings.CLIENT_BASE_URL}", params: dict = {}):
+        self.api_url: str = api_url
+        self.params: dict = params
 
-    for item in items:
-        field, value = item.split(":", 1)
-        if field in field_list_to_prepare:
-            # replace special chars
-            prepared_value = re.sub(regex, subst, value, 0, re.MULTILINE)
-            # replace multiple space
-            prepared_value = re.sub(" +", subst, prepared_value, 0, re.MULTILINE)
-            filter_prepared = (
-                field + ":" + re.sub(regex, subst, prepared_value, 0, re.MULTILINE)
+    def get(self, path: str = "/", headers: dict = None) -> dict:
+        try:
+            return super().get(path=path, headers=headers)
```

Comment:
I don't think `path` should be variable here - it is defined by the API.

```suggestion
    def get(self, headers: dict = None) -> dict:
        try:
            return super().get(path="/", headers=headers)
```

Reply:
Oh I see - this is used by the other functions. Ignore this.